components/raftstore/src/store/peer.rs

```diff
@@ -1976,10 +1980,22 @@ where
     /// Responses to the ready read index request on the replica, the replica is not a leader.
     fn post_pending_read_index_on_replica<T>(&mut self, ctx: &mut PollContext<EK, ER, T>) {
         while let Some(mut read) = self.pending_reads.pop_front() {
-            // addition_request indicates an ongoing lock checking. We must wait until lock checking finished.
-            if read.addition_request.is_some() {
-                self.pending_reads.push_front(read);
-                break;
+            // The response of this read index request is lost, but we need it for
+            // the memory lock checking result. Resend the request.
+            if let Some(read_index) = read.addition_request.take() {
```

Comment:
Can it result in too many retry requests?

Reply:
In my internal tests, this happens very rarely in practice.
@gengliqi Do you have any concern about it? Could there be some case in theory that many requests are retried?

Reply:
I believe many retry requests happen only when the network is repeatedly disconnected and connected.
So I think it's ok.